nexus/server/src/main.rs

```diff
@@ -199,79 +199,134 @@ impl NexusBackend {
                         "OK", None,
                     ))])
                 }
-                PeerDDL::CreateMirrorForCDC { flow_job } => {
+                PeerDDL::CreateMirrorForCDC {
```

Comment:
can we refactor handling this branch to its own function? It seems pretty complex now, can you refactor to make it readable.